src/main/java/com/googlecode/download/maven/plugin/internal/cache/FileIndexResource.java

```diff
@@ -0,0 +1,53 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+import org.apache.http.annotation.Contract;
+import org.apache.http.annotation.ThreadingBehavior;
+import org.apache.http.client.cache.Resource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@Contract(threading = ThreadingBehavior.IMMUTABLE)
+public class FileIndexResource implements Resource {
```

Comment:
```suggestion
public final class FileIndexResource implements Resource {
```

Reply:
Why?

Reply:
@ajarmoniuk well, there are 2 reasons:
1. Inheritance is considered to be a bad practice in general (see inheritance vs composition discussions)
2. This particular class is not intended to be inherited, thus, should be final